### PR TITLE
tgui-next: Mining Vendor

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR converts Mining Vendor to tgui-next, akin to the regular vendors. Equipment in the vendor now also counts for vending icon preview, which allows it to display it in the UI. Default marker beacon icon was also added for this purpose, as otherwise it would not show anything.

Example image:

MiningVendorNEW

## Why It's Good For The Game

Better UI.

## Changelog
:cl: Arkatos
add: Mining Vendor now uses tgui-next.
/:cl:

### CODE REVIEW

**Review by stylemistake — COMMENTED** [review]
Have you looked into the possibility of reusing the existing vending machine UI? Really the only difference is the currency, but I'm not sure if adding that extra currency bloat is worth the effort. Good stuff overall.

**Comment by Arkatos1** [comment]
I would rather keep it seperate, as it would probably end up more messy for the vending UI if I tried to accomodate it for the mining vendor. Currency is one problem, but there is also a stock count and on_station and departmental checks, which would get even more complicated with Free Golems using subtype of the mining vendor. It does not help its not even pathed as a real vendor, but a makeup one.

**Review by stylemistake — DISMISSED** [review]

File: code/modules/mining/machine_vending.dm [hunk 1]
```diff
@@ -182,6 +221,7 @@
 		new /datum/data/mining_equipment("Modification Kit",    		/obj/item/borg/upgrade/modkit/trigger_guard,					1700),
 		new /datum/data/mining_equipment("The Liberator's Legacy",  	/obj/item/storage/box/rndboards,								2000)
 		)
+	. = ..()
```
Comment (ShizCalev) [inline code/modules/mining/machine_vending.dm]: return ..()

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mining/machine_vending.dm (hunk 1 above)
Comment (Arkatos1) [reply code/modules/mining/machine_vending.dm]: Done.

File: tgui-next/packages/tgui/interfaces/MiningVendor.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  let inventory = [
+    ...data.product_records,
+  ];
+  return (
+    <Fragment>
+      <Section title="User">
+        {data.user && (
+          <Box>
+          Welcome, <b>{data.user.name || "Unknown"}</b>,
+            {' '}
```
Comment (AnturK) [inline tgui-next/packages/tgui/interfaces/MiningVendor.js]: `&nbsp;`  ?

File: code/modules/mining/machine_vending.dm [hunk 3]
```diff
@@ -71,57 +73,95 @@
...
 	src.cost = cost
 
+/obj/machinery/mineral/equipment_vendor/Initialize()
+	. = ..()
+	build_inventory()
+
+/obj/machinery/mineral/equipment_vendor/proc/build_inventory()
+	for(var/p in prize_list)
+		var/datum/data/mining_equipment/M = p
+		GLOB.vending_products[M.equipment_path] = 1
```
Comment (AnturK) [inline code/modules/mining/machine_vending.dm]: This is pretty horrible, but that's not your fault.

**Review by AnturK — COMMENTED** [review]
Switch that formatting and it looks good

File: tgui-next/packages/tgui/interfaces/MiningVendor.js (hunk 2 above)
Comment (stylemistake) [reply tgui-next/packages/tgui/interfaces/MiningVendor.js]: I'd keep using `{' '}`, because spaces should be breakable.

Comment (Arkatos1) [reply tgui-next/packages/tgui/interfaces/MiningVendor.js]: `{' '}` is also used in many other tgui-next interfaces, while there are none with `&nbsp;`  in it, so I would also rather feel more comfortable with this solution.

Comment (AnturK) [reply tgui-next/packages/tgui/interfaces/MiningVendor.js]: Alright, it's not much of a difference anyway.

**Pull request merged**